tests/infrastructure/test_data_manager_stack.py

```diff
@@ -26,3 +32,824 @@ def test_sds_data_manager_stack(app, sds_id):
+                        {
+                            "Fn::GetAtt": [
+                                Match.string_like_regexp("SDSMetadataDomain*"),
+                                "Arn",
+                            ]
+                        },
+                        '/*\\"}],\\"Version\\":\\"2012-10-17\\"}"},"outputPaths":["DomainConfig.AccessPolicies"],"physicalResourceId":{"id":"',
+                        {"Ref": Match.string_like_regexp("SDSMetadataDomain*")},
+                        'AccessPolicy"}}',
+                    ],
+                ]
+            },
+            "Update": {
+                "Fn::Join": [
+                    "",
+                    [
+                        '{"action":"updateDomainConfig","service":"OpenSearch","parameters":{"DomainName":"',
+                        {"Ref": Match.string_like_regexp("SDSMetadataDomain*")},
+                        '","AccessPolicies":"{\\"Statement\\":[{\\"Action\\":\\"es:*\\",\\"Effect\\":\\"Allow\\",\\"Principal\\":{\\"AWS\\":\\"*\\"},\\"Resource\\":\\"',
+                        {
+                            "Fn::GetAtt": [
+                                Match.string_like_regexp("SDSMetadataDomain*"),
+                                "Arn",
+                            ]
+                        },
+                        '/*\\"}],\\"Version\\":\\"2012-10-17\\"}"},"outputPaths":["DomainConfig.AccessPolicies"],"physicalResourceId":{"id":"',
+                        {"Ref": Match.string_like_regexp("SDSMetadataDomain*")},
+                        'AccessPolicy"}}',
+                    ],
+                ]
+            },
+            "InstallLatestAwsSdk": True,
+        },
+    )
+
+
+def test_log_groups_resource_count(template):
+    template.resource_count_is("AWS::Logs::LogGroup", 3)
+
+
+def test_sdsmetadatadomain_slow_search_logs_resource_properties(template):
+    template.has_resource(
+        "AWS::Logs::LogGroup",
+        {
+            "DeletionPolicy": "Retain",
+            "UpdateReplacePolicy": "Retain",
+        },
+    )
+    template.has_resource_properties("AWS::Logs::LogGroup", {"RetentionInDays": 30})
+
+
+def test_sdsmetadatadomain_slow_index_logs_resource_properties(template):
+    template.has_resource(
+        "AWS::Logs::LogGroup",
+        {
+            "DeletionPolicy": "Retain",
+            "UpdateReplacePolicy": "Retain",
+        },
+    )
+    template.has_resource_properties("AWS::Logs::LogGroup", {"RetentionInDays": 30})
+
+
+def test_sdsmetadatadomain_app_logs_resource_properties(template):
+    template.has_resource(
+        "AWS::Logs::LogGroup",
+        {
+            "DeletionPolicy": "Retain",
+            "UpdateReplacePolicy": "Retain",
+        },
+    )
+    template.has_resource_properties("AWS::Logs::LogGroup", {"RetentionInDays": 30})
+
+
+def test_iam_roles_resource_count(template):
+    template.resource_count_is("AWS::IAM::Role", 7)
+
+
+def test_s3_auto_delete_iam_role_resource_properties(template):
```

Comment:
The next 7 tests are all asserting the same thing I think. "Does the entire template have a resource with this text". I'm not sure if you can do a loop over all 7 roles to get the nested resource policy? Or is there actually only one resource policy that is the shared by all 7 users, in which case we could delete the following 6 tests.

Reply:
This is one thing I wasn't sure about. In the stack's json, these showed as 7 separate resources, but they're all exactly the same. I can condense this down to one function and a loop if you think that would make more sense.

Reply:
My thought was that even though repeating is useless when running the test, it at least makes it explicit to people using the file that there are these seven tests and the test name indicates what each of the seven roles are named even though the tests can't be mapped to a specific one since all the key, values are the same. I was hoping there was a way to test the role name against an expected string, but I haven't found a way to do that. I'll leave them as separate tests for now, but I don't have a strong opinion on this and am kind of torn myself, so I can change this if others feel differently.

I did just add properties to the lambda functions that say which role is attached to each, so at least there is a mapping of Role name and Lambda function

Reply:
I would if we could do something like `def test_make_sure_7_roles_created()` and use the `findXxx()` (with the properties) method and assert against the size of the returned set.

> Beyond assertions, the module provides APIs to retrieve matching resources. The findResources() API is complementary to the hasResource() API, except, instead of asserting its presence, it returns the set of matching resources.

Reply:
Sounds good! I'll give that a shot